src/Scientist/Internals/Experiment.cs

```diff
@@ -47,24 +82,13 @@ public ExperimentInstance(string name, Func<Task<T>> control, Func<Task<T>> cand
                 candidateResult = await Run(_candidate);
                 controlResult = await Run(_control);
             }
+            return new Tuple<ExperimentResult, ExperimentResult>(controlResult, candidateResult);
+        }
 
-            // TODO: We need to compare that thrown exceptions are equivalent too https://github.com/github/scientist/blob/master/lib/scientist/observation.rb#L76
-            // TODO: We're going to have to be a bit more sophisticated about this.
-            bool success =
-                controlResult.Result == null && candidateResult.Result == null
-                || controlResult.Result != null && controlResult.Result.Equals(candidateResult.Result)
-                || controlResult.Result == null && candidateResult.Result != null;
-
-            // TODO: Get that duration!
-            var observation = new Observation(_name, success, controlResult.Duration, candidateResult.Duration);
+       
 
-            // TODO: Make this Fire and forget so we don't have to wait for this
-            // to complete before we return a result
-            await Scientist.ObservationPublisher.Publish(observation);
```

Comment:
This is the one line I wanted to fix. Gah so much for one small functional change.